### ISSUE TITLE
[Improvement]: Allow Pimcore to work in `<iframe>` (e. g. for Cypress.io integration tests)

### ISSUE THREAD

**Issue by dominik-horb-umg** [pimcore/pimcore#15697]
### Improvement description

For our custom Pimcore code we are trying to automate some integration tests with Cypress.io, which runs Pimcore inside an `<iframe>`. This currently fails with errors like the following: `
(uncaught exception)
TypeError: Cannot read properties of undefined (reading 'settings')`. 

This is caused by the `t` JS translation function of Pimcore using `parent.pimcore` in: https://github.com/pimcore/pimcore/blob/1520bad0201d265c5faba8ac5dce006412b9a882/bundles/AdminBundle/Resources/public/js/pimcore/functions.js#L75 which usually points to the current `pimcore` JS instance because `parent` is either the current window, if there is no parent, or the parent of the current `<iframe>` ([See MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/parent)).

When run in any other `<iframe>` where `pimcore` is not set this will fail the startup.

The easiest fix I could imagine would be something like:

```javascript
if(parent.pimcore && parent.pimcore.settings.debug_admin_translations)
```

A slightly more complex solution could probably fall back to the current global `pimcore` instance if it is not present on the `parent`.

Happy to open an MR if I'm provided with some direction on the preferred solution.

**Comment by dominik-horb-umg** [comment]
Forgot to mention that there is a workaround for Cypress by setting the configuration `"experimentalSourceRewriting": true`. Since this is an experimental feature, it doesn't feel too comfortable to rely on it.

When looking at the errors shown in the closed https://github.com/pimcore/pimcore/issues/1287 I'm also not sure that the problem we found is the only issue blocking iFrame loading, but should probably be fairly easy to grep for other occurrences when the direction is clear.

**Comment by msnyder-wayside** [comment]
This is still an issue in Pimcore 10; any chance there will be a resolution?

**Comment by dvesh3** [comment]
Fixed by #15669
